system/jhtree/jhtree.cpp

```diff
@@ -878,7 +878,7 @@ unsigned CKeyStore::setKeyCacheLimit(unsigned limit)
     return keyIndexCache.setCacheLimit(limit);
 }
 
-IKeyIndex *CKeyStore::doload(const char *fileName, unsigned crc, IReplicatedFile *part, IFileIO *iFileIO, unsigned fileIdx, IMemoryMappedFile *iMappedFile, bool isTLK)
+IKeyIndex *CKeyStore::doload(const char *fileName, unsigned crc, IReplicatedFile *part, IFileIO *iFileIO, unsigned fileIdx, IMemoryMappedFile *iMappedFile, bool isTLK, size32_t blockedIOSize)
 {
     // isTLK provided by caller since flags in key header unreliable. If either say it's a TLK, I believe it.
     IKeyIndex *keyIndex;
```

Comment:
This key should also probably be dependent on blockedIOSize.  Otherwise, if a graph uses both sequential and random access to the file you will end up using too big a buffer for random access or too small for sequential.
I suspect unlikely to happen, but worth protecting against.

Reply:
agree, and the previous incarnation in Thor's IFileIO cache was based on blocksize.
I'll change it.

CKeyStore::clearCacheEntry is 'interesting':
```
            const char *name = mapping.queryFindString();
            if (strstr(name, keyName) != 0)  // keyName doesn't have drive or part number associated with it
```

probably non problematic, because of the nature of the physical filenames.. was it strstr because '/'+crc was added in distant past?